modules/admin-ui-picard/app/src/components/events/partials/DeleteEventsModal.js

```diff
@@ -0,0 +1,153 @@
+import React, {useState} from "react";
+import {useTranslation} from "react-i18next";
+import cn from "classnames";
+import {deleteMultipleEvent} from "../../../thunks/eventThunks";
+import {getSelectedRows} from "../../../selectors/tableSelectors";
+import {connect} from "react-redux";
+
+/**
+ * This component manages the delete bulk action
+ */
+const DeleteEventsModal = ({ close, selectedRows, deleteMultipleEvent }) => {
+    const { t } = useTranslation();
+
+    const [allChecked, setAllChecked] = useState(true);
+    const [selectedEvents, setSelectedEvents] = useState(selectedRows);
+
+    const deleteSelectedEvents = () => {
+        const response = deleteMultipleEvent(selectedEvents);
+        close();
+    };
+
+    // Select or deselect all rows in table
+    const onChangeAllSelected = e => {
+        const selected = e.target.checked;
+        setAllChecked(selected);
+        let changedSelection = selectedEvents.map(event => {
+            return {
+                ...event,
+                selected: selected
+            }
+        });
+        setSelectedEvents(changedSelection);
+    };
+
+    // Handle change of checkboxes indicating which events to consider further
+    const onChangeSelected = (e, id) => {
+        const selected = e.target.checked;
+        let changedEvents = selectedEvents.map(event => {
+            if (event.id === id) {
+                return {
+                    ...event,
+                    selected: selected
+                }
+            } else {
+                return event
+            }
+        });
+        setSelectedEvents(changedEvents);
+
+        if (!selected) {
+            setAllChecked(false);
```

Comment:
Currently this only unchecks the select-all-checkbox, if any of the events are unchecked, but doesn't re-check it, if by checking one of the events all in the list end up checked.

Reply:
I know that, but it still considers all events if their rows are checked. Shall I revisit this? you didn't request changes